Endless Sky 0.9.10 Merged with gles2 branch

Cat-Lady opened the pull request:
Hello Jonny,
I know it have been many years since you did the work on making Endless Sky OpenGLES 2+ compatible, and that at the time, you meet very little enthusiasm.

Time have passed, and Es is much more mature - and diverse - project now. I'm particular user that is very interested in porting the game to mobile platforms - including Android devices, Pandora/Pyra handhelds, etc. I'm fairly sure that - once brought to working stage - I can upstream it, making the OpenGLES branch part of officially maintained ES part.

I've went ahead and spent (quite long) time carefully porting your changes to up-to-date version of Endless Sky. I did my best - hoever, my OpenGLES knowledge is marginal at best, so there might be (with high probability) errors.

Still, the absolute bulk of work is done and I believe that for someone with at least baseline decent level of OpenGLES knowledge, fixing those should be trivial. If you can find motivation to review my changes and fix my newbie'ish errors, that would be really, *really* appreciated.

Cheers,
/CatLady

tehhowch committed:
Specify an initial planet for mission NPCs

e.g.
npc
    system "Sol"
    planet "Luna"

tehhowch committed:
Specify planets for NPCs in "FW Syndicate Capture"

Amazinite committed:
Fixed a Heliarch ship with negative outfit space

Fzzr committed:
Modify human warship H2H to better fit their roles

#4216

Janaszar committed:
Change Lead's description to be scientifically accurate

#4221

tehhowch committed:
Cast pre-multiplication & increase variable size

else computed results may overflow available bits resulting in UB.

 - Defer to `auto` for results of bitwise &
   - `(int)A & (uint)B` will be unsigned int/long per usual arithmetic conversions, and positive even for A < 0
 - Change from `int` to `unsigned` where negative values would result in unexpected behavior
 - When checking multiple grid cells only `diff` may be negative, so only `diff` need be signed

Fzzr committed:
Correct <source> to <origin> in syndicate prisoner jobs

#4231

Zitchas committed:
Fix missing quotation mark in Remnant:RS2 (#4234)

* Update remnant missions.txt

Missing quotation mark at the end of line 879, the end of Remnant: RS2.

* Update remnant missions.txt

Missing comma in RLS1 and  RCR.
Capitalizing Captain in RS1.

(courtesy of Pulsar)

Amazinite committed:
Update changelog (#4207)

* Update changelog

* Changed tabs into spaces

* Typo fix

* Typo fixes

* Wording changes

* Slight changes

* Update changelog

* Big changes

* Update changelog

* Add more changelog updates

oo13 committed:
Fix a bug for carriers to get lost. (#4200)

tehhowch committed:
Update changelog

beccabunny committed:
New lagrange image (#4226)

* Delete station0.png

* New lagrange

* Update copyright

tehhowch committed:
Cap projectile velocities to 450000

 - Including when fired from very fast ships, or through accelerating submunitions, etc.
 - Prints a one-time (per session) warning about the speed violation

comnom committed:
Parse GameEvents for additional definitions (#4238)

* Parse events for additional defines
* Simplify deferred definition loading
  - Use const & for `key`
  - Remove need for if-else if chain.

This allows GameEvents to be the sole source of certain game object definitions without generating spurious warning messages.

tehhowch committed:
Update changelog

tehhowch committed:
Check for user-supplied unrepresentable values

Current allowed condition values are those of signed int (generally -2'147'483'648 to 2'147'483'647)

tehhowch committed:
Use 64-bit int for ConditionSet

thus allowing modders to safely use rather large values in conditions

tehhowch committed:
Use 64-bit int for PlayerInfo::conditions

thus allowing rather large conditions to be serialized.

This is particularly useful for financial conditions. No change to the combat rating limit of 2b is planned.

tehhowch committed:
Fix possible overflows for high-income players

tehhowch committed:
Update changelog

tehhowch committed:
Remove extra power-of-two

tehhowch committed:
"Fail" audio no longer plays on loop

 - if you have no manner of jumping
 - if you cannot reach your target system

Behavior is now consistent with having insufficient fuel to jump.

Refs endless-sky#3633
Refs 65b70c02db61f7c835eb8ed4e3b67383ea0339f8

tehhowch committed:
Use singular or plural credit amount

 - Always include credit/credits for simplicity

e.g. "You paid 100 credits in crew salaries, 1 credit in mortgage payments, and 1,234 credits in fines."

tehhowch committed:
Refactor the check for planet accessibility

 - Pre-strip the "requires: " prefix from required attributes
 - Simplify test for "do we even need to look at the ship's attributes?"

tehhowch committed:
Enforce required attributes for in-flight spawns

Otherwise, normally specified "system fleets" could take off from a planet even when they do not have the capability to land on it.

Pointedstick committed:
Tweak and differentiate the starting ships (#4219)

* Give the Star Barge an anti-missie turret by default and tweak it accordingly

* Tweak the Shuttle, Sparrow, and Finch as well

* Tweak the tweaks

* Reduce the magnitude of some changes, and undo others

* Revert Shuttle drag change

beccabunny committed:
New Pug Gridfire Turret and T3 anti missile image (#4196)

* Update pug.txt

* Update copyright

* Add files via upload

* Add files via upload

* Update pug.txt

* Delete pug gridfire turret.png

* Add files via upload

* changed description for the turret (thanks jafdy)

* Added T3 AMT

* Add files via upload

* Delete pug gridfire turret.png

* Uploaded image to fit 2x

* Update pug.txt

* Delete pug gridfire turret.png

* Add files via upload

* Update copyright

* Description correction

* Delete T3 anti missile.png

* Updated image for better resolution

* Update data/pug.txt

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update copyright

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update copyright

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update copyright

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update copyright

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update data/pug.txt

Co-Authored-By: beccabunny <45330165+beccabunny@users.noreply.github.com>

* Update copyright

* Delete T3 anti missile.png

* Delete pug gridfire turret.png

* Better hardpoints

* Update pug.txt

* Delete T3 anti missile.png

* Removed capitalization

* Delete T3 anti missile.png

* Add files via upload

* Update pug.txt

* Update copyright

* Update copyright

* modified description

* Update pug.txt

* Update pug.txt

* Update pug.txt

* Update pug.txt

tehhowch committed:
Remember the name used for a news portrait

tehhowch committed:
(Re)Allow multiple news items per landing

but require pressing the key again to get additional news.

tehhowch committed:
Specify whether KeyDown event was a repeat or not

 - Cast the `repeat` Uint8 field of SDL_KeyboardEvent to bool
   https://wiki.libsdl.org/SDL_KeyboardEvent

tehhowch committed:
News entries should not nest double-quotes

as the text is already presented as a quotation from the person. Instead, use single quotes.

Brick63 committed:
Add other artist names to the credits (#4232)

* Add Beccas name to credits

they have added art so seems to be appropriate

* add in the other names

beccabunny committed:
Correct osprey sprite (#4244)

* Delete osprey.png

* Add files via upload

* Delete osprey.png

* Add files via upload

* Delete osprey.png

* Updated size

tehhowch committed:
Improve landing message handling in AI::MovePlayer

 - Once the player starts hyperspacing, do not print landing messages (since they will be for the to-be-departed system)
 - Require that the player's StellarObject target is in the player's current system (instead of possibly being in the destination system)

tehhowch committed:
Add additional comments to AI class

tehhowch committed:
Ensure that hostile escorts are treated by `n` appropriately

tehhowch committed:
Update changelog

comnom committed:
Add 128x and 512x icon sizes to SConstruct installs (#4248)

refs: #3913

Commit:
Made it so the Terraforming 8 continuation mission can only be offered in human space.

Commit:
Fixing a mistake in my previous commit.

tehhowch committed:
Explicitly check for the "gun ports" attribute

Otherwise installing outfits with `weapon` nodes would result in equipping them as a gun

Refs endless-sky#4251

Amazinite committed:
Reduced difficulty of finding missions to earn the Bactrian

#4253

Zitchas committed:
Added a Remnant scanning outfit

#4211

Janaszar committed:
Spelling & Grammar Fixes

#4255

Zitchas committed:
Replacing Tali with Taely

#4254

Zitchas committed:
Added the Remnant scanner to the outfitter

#4263

tehhowch committed:
Add missing ship reference for offered assisting/boarding missions

tehhowch committed:
Update changelog

Commit:
Tefkar Ret station no longer fines the player

#4265

Zitchas committed:
Changed "airport" to "spaceport" in Wanderer mission

#4267

CyberShadow committed:
Improve dynamic zooming (#3727)

* Improve dynamic zooming

Behavior changes:

- Fixes the view zoom setting getting reset if the window is initially
  created at a smaller size (a likely occurrence when using tiling
  window managers)

- Makes the view zoom restriction dynamic, updating the zoom level as
  necessary in real time as the window size changes, up to the user
  preference

- Updates the view restriction bounds, taking into account both the
  window width and height, to ensure that at least the main menu and
  preferences panel are fully visible and accessible

Code changes:

- Split the "view zoom" value into two logical values:

  - User zoom, the setting that exists in the configuration file and
    is visible in the configuration panel

  - Effective zoom, used to actually scale the UI - a function of user
    zoom and window size

- Update the code to use the user zoom or effective zoom values as
  appropriate

- Consolidate all view zoom restricting code (from main.cpp and
  PreferencesPanel.cpp) into one place (Screen.cpp)

* Revert Zoom->EffectiveZoom renaming
& remove duplicated zoom reset

tehhowch committed:
Cleanup & add comment

 - If `zone.Value()` is one zoom setting, it is not the other.
 - Restored trimmed tab
 - Update changelog

Zitchas committed:
Changing OK to Okay (#4270)

Updating to use the writing style of ES.

thomasballinger committed:
Improve interpolation between game zoom levels (#3828)

* Make `log(zoom)` lerp to `log(targetZoom)`
* Only compute zoom change when needed

MCOfficer committed:
Grammar/spelling fixes in Remnant missions

#4266

comnom committed:
check carried ship's parent system in NPC::HasSucceeded

refs: https://github.com/endless-sky/endless-sky/issues/4281

comnom committed:
Made FW Epilogue missions minor

#4282

Zitchas committed:
Reworked the Remnant Tech Retrieval missions

#4269

comnom committed:
Edit the human AMTs for better visibility

#4250

comnom committed:
Lowered the scan sfx pitch by one octave

#4252

beccabunny committed:
Added color to the Pug galaxy image

#4256

comnom committed:
Attach to / allocate a console output on Windows (#4249)

* Add windows console output
* Perform stream redirections even when allocating a console
 - Otherwise the "stream > /dev/null" behavior persists

Amazinite committed:
Fixed the style of some choices from #4269

Commit:
Updating the version number.

Amazinite committed:
Added more options to increase pirate rep through pirate jobs

#4287

Amazinite committed:
Update changelog

#4291

Amazinite committed:
Changed the contribution notice in the credits

Moved it to the top so that it's easier to notice.
Added reference to reporting bugs and giving feedback.
Added reference to the Steam forums and Discord server, as those are also places where we take feedback and bug reports (which we funnel to GitHub regardless if we find an actual bug or get good feedback).
Ref: #4295

Amazinite committed:
The Deep: Interrogation mission now properly declines

comnom committed:
Bugfix: Terraforming 6 should not offer in Sol (#4296)

If the source for Terraforming 6 is in Sol, the deadline becomes 0, making it impossible to complete.

Reported by Kiko the Exile on discord.

comnom committed:
Added a landing message for black holes

#4297

Zitchas committed:
Changed the Remnant S-970 mission's description to refer to the correct outfit

#4299

ZBok committed:
Pirate occupation missions now accept instead of launching

#4300

Amazinite committed:
Explicitly state the hit force of the Hai Tracker as being a feature

#4298

comnom committed:
Add nullptr check for 87981a4

Zitchas committed:
Fix Typo (#4322)

developped -> developed
Spotted by @Darcman99

Amazinite committed:
Extended the Terraforming 6 deadline by 2 days

This avoids the mission from offering so close to Sol that the deadline gives you no wiggle room.

Amazinite committed:
Pug and FW are now properly hostile during FWC Pug 3B

#4309

Amazinite committed:
Various Deep missions now check to make sure that the war is not ongoing

#4311

Amazinite committed:
Reduced the speed of the Centipede

#4316

Zitchas committed:
Remnant Tech Retrieval missions are no longer marked minor

#4331

Zitchas committed:
Typo fix (#4332)

Spotted by @Proont

tehhowch committed:
Reduce excessive damage from Timer Ship

 - Remove distance-based scaling from Timer Weapon
 - Enable Timer Ship to fire an unsafe weapon at an unsafe range

Closes endless-sky#4271

tehhowch committed:
Remove unneeded "Timer Submunition" attributes

Zitchas committed:
Typo fix

#4341

Zitchas committed:
EMP ammo now requires a license to purchase

#4338

Amazinite committed:
Player will now receive on visit dialog for cargo and passenger jobs

#4315

Amazinite committed:
Corrected the descriptions of the Sidewinder and Meteor missiles

Zitchas committed:
Fixed references to the Remnant: Plasma Cannon mission

#4353

Zitchas committed:
Clarified a sentence in a Remnant mission

#4354

Amazinite committed:
Made Terraforming 6 not offer on uninhabited worlds

Luckz committed:
README now links to the game's Steam store page rather than the community page

#4360

Zitchas committed:
Normalized capitalization of ship and outfit names in the Remnant missions

#4363

Zitchas committed:
Added a missing quotation mark

#4359

Amazinite committed:
Mystery retrieval job now mentions the source/destination in its description

Amazinite committed:
The Navy ships in Mystery Cubes 2 now have the proper government

Amazinite committed:
Gave Tactical and Asteroid scanners to ships where appropriate

#4367

starscollide5 committed:
Reduced the required crew of the Marauder Splinter to better match the base Splinter

#4373

Ref: 7eb953a

jostephd committed:
Add missing `min` when autorefilling ammunition (#4365)

Don't install from stock more copies of the outfit than the ship can
receive.

Closes endless-sky#4364

tehhowch committed:
Add additional sanity check when refilling ammo

tehhowch committed:
Fix non-installed outfits showing depreciated plunder value.

Ref: 88eed52
Closes endless-sky#4317

Amazinite committed:
Added an extra mission to Wanderer middle so the player has one active after the Sestor Remnant

#4324

Amazinite committed:
Added on visit dialogs for nearly all missions that can trigger them

#4342

Amazinite committed:
Added missile storage outfits to ship missile variants and ships that largely use missiles

#4371

Amazinite committed:
Error fix

Amazinite committed:
Error fix

Amazinite committed:
Added shipyard thumbnails for the remaining ships that didn't have them yet

#4377

warp-core committed:
Fix branches in Remnant tech missions

#4383

Amazinite committed:
Generic jobs can now offer from independent worlds

#4328

Amazinite committed:
Made depreciation happen about 3 times slower (for real this time) and added a 7 day grace period before it begins

Ref: 1d0fe94
#4372

warp-core committed:
Fix text replacement in Deep: TMBR 0

#4389

MCOfficer committed:
Typo fix

#4390

Amazinite committed:
Fixed the world-ship thumbnails

The post-processed shadowing is different because the assets didn't save properly and I had to redo them. The link to the assets in the PR has been updated accordingly.

Amazinite committed:
Merge branch 'master' of https://github.com/endless-sky/endless-sky

tehhowch committed:
Fix failed OSX CI builds

that were failing because the VM's "Homebrew" installation is out-of-date.

 - Increases the `update: true` indentation so it is actually applied
 - Cache downloaded Homebrew resources (unfortunately doesn't speed up `brew update`)

tehhowch committed:
Add more eligibility criteria to NPC-carried outfits

While it's possible for Outfits Expansions, Bunk Rooms, and similar to be carried in cargo, it does look strange.
This commit prevents NPCs from carrying randomly picked outfit cargo that
 - modifies "bunks"
 - increases outfit space
 - increases cargo space

Refs endless-sky#4321

tehhowch committed:
Bugfix: Limit CargoHold::Transfer when size = 0

Fixes endless-sky#4305

A non-negative CargoHold::size means the hold has a limited capacity that should be respected.

tehhowch committed:
Require crew for certain assist/board missions

where the conversation suggests/requires a captain of some kind.

Closes endless-sky#920

Amazinite committed:
Tweaked wait times in Deep missions 

Feedback suggests that the wait times were too long, trivializing the later combat by allowing players to build up too big of a fleet. 
Event timers that had a range now only use set times. Generally reduced event timers.
Previous max wait time: 278 days.
Previous min wait time: 186 days.
Current wait time: 137 days, making wait times anywhere from 49 to 141 days shorter.

oo13 committed:
Escorts refuel more reasonably when they have both a hyperdrive and a jumpdrive. (#4288)

* Escorts refuel more reasonably when they have both a hyperdrive and a jumpdrive.

* Add additional comments

ChamEV committed:
Update system colors when switching between Shipyard and Outfitter map panels (#4425)

The cached map state wasn't being updated.

petervdmeer committed:
Fixed minor code style issues (#4428)

petervdmeer committed:
Reorder some includes to match style (#4430)

petervdmeer committed:
Some more style fixes. (#4429)

No big changes, just some spaces removed.

ChamEV committed:
Reordered drawing on map mission panel so rings are behind UI elements. (#4416)

Fixes endless-sky#4415

AlexBassett committed:
Pirate occupation missions will no longer have Humanika as a destination

#4433

ChamEV committed:
Removed dead code, these buttons are handled by the interface class. (#4401)

beccabunny committed:
New Southbound ship sprites (#4319)

* Detailed sprites

* updated copyright

* better blur on the fury

* Redid all the normal dpi scaling

* Added the new Bastion with the new hardpoints position

* Added Bastion to copyright and corrected an error

Switched from " " to "?"

Zitchas committed:
Change Bactrian Classification (#4387)

This PR is purely cosmetic, and is an effort to redirect the perception and fixation on the Bactrian by making one simple change.

1) The Bactrian is reclassified as a Transport.

ChamEV committed:
Moved done button on map to far right, adjusted button sizes. (#4402)

* Moved done button to far right on map screen, click areas match button visible areas.

* Added 14 extra pixel separation between done button and other buttons (24 total).

petervdmeer committed:
Really minor code comment tweaks (#4400)

* Minor code comment improvements

Minor improvements to code comments.

* Made some comments more clear

Signed-off-by: Peter van der Meer <peter.vd.meer@gmail.com>

ChamEV committed:
Buy, Sell, Leave button size fix (#4351)

* Adjusted Buy, Sell, Leave buttons clickable and hover area in outfit and shipyard shop to match the displayed button size.

* Removed unneeded lines/comments

* Fixed tab/space.

* formatting.

* Made buttons clickable one pixel wider to the right.

Amazinite committed:
Revert "Change Bactrian Classification (#4387)"

This reverts commit 919e6b4597d05f3b6689520fe24a42637f2c6f99.

Amazinite committed:
Clarified some writing in Paradise Fortune 4

Amazinite committed:
Fixed a missing quotation mark in the previous commit

beccabunny committed:
Updated the boxwing's copyright (#4440)

* updated the boxwing's copyright

* removed excessive space

tehhowch committed:
Add scrolling to MissionPanel job/mission panes (#4424)

* Add some additional comments to MissionPanel

* Add preliminary automatic scroll updating when handling keyboard input

* Improve autoscroll behavior

* Scroll the respective mission list when clicking names or systems

* Draw mission lists before the map legend and selected system info
so that a list overflowing the available area will always be "behind" other UI elements

* Switch from paginated scrolling to "smart" scrolling
i.e. scroll only the minimum necessary, and only when necessary.

* Use a snap-to scroll instead of interpolated

Zitchas committed:
Spelling & Grammar in line 695 (#4452)

Fixes:
- Changes obslete to obsolete
- Capitalizes "She"

Thanks to 2214441811999913317771666 on the discord server for pointing out the former and bringing our attention to this section.

Thanks to Epsilon Carinae on the discord server for pointing out the lack of capitalization.

realityforge committed:
Fix several missions that seem to have lines that have an extra tab in the definition

#4461

LocalGod79 committed:
Updated Marauding Max

#4448

sdennie committed:
Change aim to be explicitly signed (#4457)

Some architectures default `char` as unsigned, so turrets would only rotate in one direction. 

Fixes endless-sky#4456

tehhowch committed:
Fix invalid Phrase token

rugk committed:
Do not translate game title (#4454)

The game title is a proper name and should not be translated.

Fixes https://github.com/endless-sky/endless-sky/issues/4453

MCOfficer committed:
Typo fix

#4470

SolraBizna committed:
Fix "Wanderers Evacuation 1C" offer condition

#4476

Commit:
Fixed the ordering of ship hardpoints after 2e2ab8c

#4474

Kryes-Omega committed:
Added missing attributes to Bloodsea

#4479

beccabunny committed:
Less blurry Southbound ship sprites

#4455

Nams-2 committed:
Removed the license from the Remnant Key Stone

This avoids a soft lock if the player finds themself without a JD or Key Stone in the Ember Waste before they're able to purchase Key Stones from the Remnant. This fix should only be considered temporary until a mechanic like in #2911 is added and the game can recognize if the player is in this state and properly rectify it.
#4491

Amazinite committed:
Unwanted Cargo now correctly identifies when you have the Hai translator

petervdmeer committed:
Fixed "Paradise Fortune 1" on visit

#4490

beccabunny committed:
Redesigned Betelgeuse Ships

#4458

beccabunny committed:
Improved Graviton engine flare

#4495

Nams-2 committed:
Change the government of the Ravens in "Deep Archeology 3"

#4497

Amazinite committed:
Fixed the Marauder Firebird (Weapons)'s hardpoints

#4500

beccabunny committed:
Corrections to Betelgeuse ships

#4501

petervdmeer committed:
Minor spelling fixes (#4504)

Fixes minor spelling error in datafile and in comment.
Typo on data/hails.txt was reported by Hyugat on discord in endless-sky-chat.

Zitchas committed:
Correcting hanger to hangar

#4505

Zitchas committed:
Two spelling fixes

#4506

Amazinite committed:
Fix extra restriction on bank accessibility (#4509)

Fixes #4508

Nams-2 committed:
Changed bounty names and descriptions to better highlight difficulty differences

#4507

Amazinite committed:
Fixed a typo that was preventing cargo text replacements in some pirate jobs

Amazinite committed:
Updated OutfitInfoDisplay to support units for attributes

Amazinite committed:
Added an autosave on FW Katya 2

This allows players struggling with Liberate Kornephoros to go back to an autosave from just before they officially join the FW and become hostile to the Navy.

tehhowch committed:
Restrict reparenting of mission-affiliated carried ships to the same missison (#4468)

* Restrict reparenting of mission-affiliated carried ships to same-mission ships
* Avoid nullptr dereference
 - Carried ships should properly inherit their parent's system, but on the off chance one doesn't, default it into the players system
* Replace std::find_if with a Mission class method
 - explicitly declare the lambda captures for `reparentWith`
* Log the issuance of a fallback system

Fixes #4307

Amazinite committed:
Update changelog for v0.9.10 (#4515)

MageKing17 committed:
Save thumbnails, just as we save sprites. (#4527)

* Save thumbnails, just as we save sprites.

This will allow variants with different thumbnails to have their thumbnails be preserved through a save/load cycle. Fixes #4503.

* Update changelog.

Amazinite committed:
Reduced the price of the asteroid scanner to make it more appealing for new pilots

A Sparrow start pilot who's just dropped off James can sell their shield generator and immediately buy an asteroid scanner at this price. After about a week of mining asteroids just above Hestia, they can pay off the asteroid scanner and begin making profit.

Amazinite committed:
Update changelog

Zitchas committed:
Updating Protector Description

#4542

Amazinite committed:
The " Kor Mereti " government now uses the Korath language

Amazinite committed:
Changed the Cosmic Devil from never disabled to uncapturable

Makes the fight a little easier and allows you to steal its technology. Still can't be captured in order to fit with the kill dialog saying that the ship explodes.

Amazinite committed:
Fixed some errors in one of the friendly pirate hails

Some combinations could result in two words smashed together or double punctuation at the end.

RestingImmortal committed:
Fix typo (#4545)

* Fix typo

second to seconds

warp-core committed:
Typo fix (#4548)

Amazinite committed:
Made the Cosmic Devil capturable and changed the dialog to fit

MageKing17 committed:
Copy requiredOutfits in MissionAction::Instantiate()

#4555

Commit:
Updating the version number.

Tadrix committed:
Altered the base attack and defense of Coalition and Heliarch crew

#4168

Zitchas committed:
Rebalanced the Pelican to allow it to reach greater speeds

#4356

Amazinite committed:
Changed the Quicksilver (Mark II)'s outfits as to not match the base model

#4496

Thunderforge committed:
Separating a list of contraband into its own phrase to provide more variety in hail messages

#4534

MageKing17 committed:
Make "Southbound Shipyard's" plural in Fury description. (#4560)

According to every other ship description, "Southbound Shipyards" is the correct name, and this was probably a transposition typo.

Nams-2 committed:
Randomly generate the offer dialog for Courier 2

#4536

Amazinite committed:
Fixed Raid on Merchants job descriptions

Zitchas committed:
Remove formatting differences in license.txt (#4393)

* Create LICENSE.md

This change simply creates license.md to provide the automated license parsing & display by GitHub. Contents are verbatim that of license.txt

(Based on MCO's explanation, GitHub specifically for LICENSE or LICENSE.md, not license.txt)

* Removing file ending from LICENSE

Removes ".md" from end of file to test the license on my fork.

* Replacing < and > with { and }

* Delete LICENSE

* Replacing license.txt with LICENSE

* Restore filename

tehhowch committed:
Allow restriction of scanning / fining by governments (#3187)

* Governments store a filter describing where they enforce laws

  - The player can be fined or scanned in any system or planet that matches one of the enforcement zones.
  - If no zones are defined, then the government has universal jurisdiction

* Only ships or planets with enforcement authority perform scans

Zitchas committed:
Tweaks to some human transport ship stats to make them more viable choices

#4569

beccabunny committed:
Heavy shuttle image fix

#4573

tehhowch committed:
Fix std::out-of-range when checking scan permissions

 - Compute scan permissions for each government, even if they aren't in the player's system yet

Zitchas committed:
Replacing all instances of datapad (#4574)

As per Amazinite's suggestion in order to maintain consistency with how it is used in the Rim archeology missions, this PR replaces all currently existing instances of "datapad" with "data pad"

Bladewood committed:
Added missing source to Terraforming 7

#4576

tehhowch committed:
Add a PR template (#4568)

* Initial Draft

Current GitHub support for multiple PR templates is lacking (url parameters only!) so people will just need to remove the bits that don't apply.

* Add high-dpi plugin PR to the Content PR checklist

Amazinite committed:
Made the source for various missions explicit instead of being implicit based off of the previous mission

Amazinite committed:
Fixed Pact Recon missions from offering on pirate planets

Pirate worlds were given regional attributes, meaning these missions could spawn on southern pirate worlds.

Zitchas committed:
Adding foreshadowing to Remnant: Void Sprites 3 for Remnant: Face to Maw

#4562

Commit:
Typo fix

#4590

Hyugat committed:
Fitted the Mark II Frigate with atomic thrusters to match the other Navy ships

#4577

wjp committed:
Fix typo in bounty missions (#4593)

Nams-2 committed:
Fix incorrect substitution target in Courier 2 (#4595)

`<source>` is not valid; the proper tag is `<origin>`

Amazinite committed:
Avoid referring to the player's standard sidearm as a laser pistol

Other missions either don't specify or suggest that the player's sidearm is a normal gun.

Zitchas committed:
Corrected the swizzle of the " Drak " government

#4597

ZBok committed:
Added some ships to the respective pirate fleets of their region

#4293

Zitchas committed:
Remnant Tweak: Giving Albatrosses and Pelicans Scram Drives

#4580

petervdmeer committed:
Added a mission where the player can get scammed

#4589

Zitchas committed:
Added advanced pirate shipyards

#4578

Amazinite committed:
Added missing hails to alien governments

#4553

Pointedstick committed:
Fix formatting on one of the new Heliarch hails

Amazinite committed:
Updated the intro with new information

#4547

Amazinite committed:
Better differentiate various human medium warships and knock the Mule down a peg (or two)

#4581

Brick63 committed:
Added an offensive Pug H2H weapon

#4563

Brick63 committed:
Added Heliarch H2H weapons

#4564

Amazinite committed:
Lowered the cost of some alien H2H weapons to better reflect their utility

MageKing17 committed:
Improved the descriptions of the Marauder ships

#4572

Amazinite committed:
Fixed some Mule variants having negative weapon space due to the recent changes

Amazinite committed:
Merge branch 'master' of https://github.com/endless-sky/endless-sky

Commit:
Added shield and hull repair outfits specifically for Heliarch ships

#4607

Amazinite committed:
Updated copyright for the recent commit

Cat-Lady committed:
Merges for bounty-fix and boarding-enhancements components that need to alter vanilla files.

Cat-Lady committed:
Synced with upstream changes of marauder's descriptions and Marauder Arrow stats.

beccabunny committed:
Fixed the Behemoth sprite

#4611

Cat-Lady committed:
Merge remote-tracking branch 'upstream/master' into captains-edition

Zitchas committed:
Creating the Anvil Class Engine

#4594

Zitchas committed:
Enabling the player to demand tribute from the Remnant

#4273

Brick63 committed:
New Laser Rifle image

#4605

Commit:
Added the hyperdrive to Efreti outfitters

#4619

Cat-Lady committed:
Merge remote-tracking branch 'upstream/master' into captains-edition

Cat-Lady committed:
Updated marauders.txt as per Boarding Enhancements

Cat-Lady committed:
Laser rifle image changed to modified Quarg Block's one.

oo13 committed:
fix: Add missing tooltip for "movement:" (#4621)

Amazinite committed:
feat: Uninhabited planets do not fine by default (#4512)

* Uninhabited planets now never fine the player

* Error fix

* Removed security 0 from Ruin

* Added a flag to check if the security of an uninhabited planet was changed

* Spacing

* Name changes

* Improved readability

Amazinite committed:
feat: Implement maintenance and operating cost attributes  (#4596)

* Added a maintenance cost outfit attribute

* Maintenance costs will now show in the bank panel

* Allow overdue maintenance to be paid off

* Properly size the max rows if you have overdue charges but no crew or maintenance

* Simplified output

* Added an operating costs attributes

* Overdue maintenance costs now impact credit score

* Pay maintenance costs after crew salaries

* Include the maintenance costs of in cargo outfits

* Comment correction

* Clean up

* Code simplification

* Check for > 0 credits instead of non-zero credits for the two crazy people who give themselves negative credits

* Check in cargo outfits when landed and prevent net negative maintenance costs

* Prevent negative costs for each ship and each outfit in cargo instead of preventing a total net negative.

tehhowch committed:
fix: Specify the `std::max` overload to use

Cat-Lady committed:
Merge remote-tracking branch 'upstream/master' into captains-edition

Cat-Lady committed:
Reverted accidentally incorporating *old* marauders, again.
This reverts commit 409ce5858b4f5287e7bb483cfa0c06909cfc9a40.

Cat-Lady committed:
Properly (this time) integrated marauder's BE.

Brick63 committed:
Added a Wanderer dedicated transport ship to be used later

#4102

tehhowch committed:
fix: mixed whitespace

tehhowch committed:
ci: revert removal of executable permission for test script

tehhowch committed:
feat: Support algebra in ConditionSet expressions (#3128)

* Handle assignment operations inside ConditionSet::Test

 - Add boolean test/apply indicator to each expression
   - Determine if a ConditionSet performs tests or assignments in the expression constructor
 - If the ConditionSet is not pure (mixes assignments and tests), use the assignment operators to alter an initally-empty map)
   - When obtaining the value from strings (e.g. conditions), first check the created map, then fall back to the player's conditions.
 - Warns when loading if an and/or node uses assignment operators

* Create classes to manage complex condition expressions

 - SubExpressions represent each side of an expression, with methods to allow serialization and evaluation
 - SE::Operation class manages the details of math within a given side, by tracking the location of operands and operator

* Implement sequence generator algorithm based on Dijkstra's shunting yard

* Alter Add methods to use SubExpressions and Operations

 - Add SubExpression and Expression constructors for simple conditions
   - Simple conditions have only one token per side, so vector constructs are unnecessary.

* Simplify adding operators, reduce code duplication

 - Add modulus to supported simple operations
    e.g.	`year % 7 == 1`
      will be true for years 1, 8, 15, ...
 - Moved the precedence map to an anonymous function
   - Simplify the process of updating supported simple operators, as all maps/sets are now in the same region of the file.
 - Reduce duplication with 'AddOperation' private method

* Extend DataNode with static methods for string parsing

* Simplify FW:Intro offer checks

* Use stdlib to simplify, update initializations

* Maintain support for legacy `never`

* Simplify warnings & check for leading adjacent (

* Tweaks

 - Avoid taking reference to `map::find` iterator (as find is return-by-value)
 - Add missing `vector::reserve()` call
 - Reduce number of different `vector::emplace_back` overloads used
 - Reduce calls to `DataNode::IsNumber` by checking for `"random"` first

* Simplify "deep convoy" checks

* Add bounds-checking for user-input condition values

 - No bounds checks done for the results of computations

* Update some comments

* revert accidental "has" -> "not" change

Cat-Lady committed:
Merge remote-tracking branch 'upstream/master' into captains-edition

Cat-Lady committed:
Fixed Marauder tacarmed variants

Cat-Lady committed:
Experimental OpenGLES merge. Might have errors. Might be not working at all. Should be very easy to fix for someone with at least decent OpenGLES skills. In any case, most of the mudane, base work is done.

Cat-Lady committed:
minor fixes

Zitchas committed:
Four new Remnant missions

#4222

JonnyH commented:
Wow, this takes me back :) I haven't played Es in years.

getting this running on the pandora (or soon(?) to be pyra) was one of my main goals. Though that does mean restricting ourselves to gles2, while today I'd normally state that gles3 was a better baseline - though I doubt this *needs* the features, I find it a bit cleaner and a much better match to "modern" desktop OpenGL - as it's pretty much a direct subset, while gles2 still requires you to do some things that are considered 'deprecated' now.

I suspect a rebase may be a better way of dealing with this changeset than a merge, as there seems to be quite a few changes needed in the merge commit.

I might be able to have a look at things this coming weekend if it's not already sorted by then.

Amazinite committed:
Fixed the offer condition of the newest Remnant mission

Zitchas committed:
Added a new Remnant carrier along with a fighter and drone

#4546

Zitchas committed:
Fixing Penguin Gun & Turret placement (#4627)

Fixed the Penguin Gun & Turret Placement

Cat-Lady commented:
Hey Jonny,
A blast from the past indeed, isn't it?

Thanks for your response! Yes, you're right - rebase would be a better choice, and indeed I am sure that there are errors on my work, as my OpenGL(ES) knowledge is subpar, at best. As said earlier, I won't hide the fact that I hoped to interest you in it, and decided that actually doing as much work as I am able too myself is best way of picking up interest in fixing the rest ;)

I'll try to do the rebase approach before weekend - unless, you plan to do it in OpenGLES 3, indeed? In such case, is there any worth in me (probably messy, in some parts) re-applying the old OpenGLES 2 work on top of current codebase?

Cheers, and I really hope that we can make it a reality,
/CatLady

PS.

Endless Sky have really gone ahead along the years, and there is much very high quality work available for it. Maybe you will find renewed interest in the game, cause it is something special (quality) in the FOSS gaming world, and I say it after having rather vast experience with various projects over the years (sadly, none of it regarding OpenGL-OpenGLES conversions... So far.)

Amazinite committed:
Implemented the Riptide into the Wanderer campaign

#4626

Amazinite committed:
Lowered the weapon capabilities of the Petrel

The Petrel had far too much weapon space compared to other fighters. With this reduction, the Petrel maintains its positions as having the most weapon space (as well as the most weapon hardpoints), but there isn't as absurd a gap. If new fighter categories existed, then the Petrel's stats could be reverted and it could be moved into the new category.

Cat-Lady committed:
Merge remote-tracking branch 'upstream/master' into opengles
